packages/react/src/components/List/ListContent/ListContent.jsx

```diff
@@ -257,10 +262,33 @@ const ListContent = ({
       ...(!hasChildren && item.hasLoadMore ? [renderLoadMore(item, isLoadingMore, level)] : []),
     ];
   };
+  const previousExpandedIds = usePrevious(expandedIds, expandedIds);
+  const notifyOnLastExpansionChange = (itemId) => {
+    // get the lastId of the diff between previous and current, this handles closes
+    const [previousLastId] = previousExpandedIds
```

Comment:
We should probably create the diffs outside the loop in order to not upset Big O

Reply:
🤦 duh. corrected. Thanks!